src/Home/useBluetoothStatus.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "react"
+import { Platform, AppState } from "react-native"
+import { isBluetoothEnabled } from "../gaen/nativeModule"
+
+export const useBluetoothStatus = (): boolean => {
+  const [btStatus, setBTStatus] = useState(false)
+  useEffect(() => {
+    const fetchBTEnabled = async () => {
+      const status = await fetchBTStatus()
+      setBTStatus(status)
+    }
+
+    AppState.addEventListener(getOSListener(), () => fetchBTEnabled())
+    return AppState.removeEventListener(getOSListener(), () => fetchBTEnabled())
+  }, [])
+
+  return btStatus
+}
+
+const fetchBTStatus = async (): Promise<boolean> => {
+  const status = await isBluetoothEnabled()
+  return status === true || status === "true"
+}
+
+const getOSListener = (): "focus" | "change" => {
```

Comment:
Again for my own education, thoughts on

`const getOSListener = (): "focus" | "change" =>`

vs

```
type ListenerMethod = "focus" | "change"

const getOSListener = (): ListenerMethod =>
```

Reply:
Yes! I like both these suggestions, thanks!